Municipality/azc.mod2g

```diff
@@ -1,13 +1,20 @@
-use dummy as knowledge.
-
-order = linearall.
-module azc{
-	%Check if there is still need for an azc (id 650)
-	if bel(building(BuildingID,_,OwnerID,_,_,650,_)) then exit-module.
-	%Check if we have budget for an azc
-	if bel(money(MoneyAmount),MoneyAmount<100000) then exit-module.
-	%Check if there is space for an azc
-
-	%Adopt if all preconditions are met
-	if true then adopt(buildAZC).
+use dummy as knowledge.
+
+order = linear.
+module azc{
+
+	%if we already have a goal for building a specific azc we exit.
+	if goal(buildAZC(_,_)) then exit-module.
+
+	%Build an azc that has the right size or is smaller (minimum 40 units)
+	if bel(
+		azc(Azc),%Azc>0,%get the amount of units we are short
+		my_stakeholder_id(StakeholderID),%we need to own the land
+		land(LandID,Stakeholder,MultiPoly,Zones,Area),%find land
+		%Area>800, Area<20*Azc+1000,%land may not be to big or small
+		member(Zone,Zones),%find zone
+		zone(Zone,_,_,_,Categories),%find zone category
+		member('SOCIAL',Categories),%zone should support function
+		money(X), X>1200*Area)%have the required money to build this azc
+		then adopt(buildAZC(LandID, 1)).%floors is calculated

```

Comment:
Hey Diony, I think this styling is a little unreadable without syntax highlighting, I would prefer not having comments in line like this, I think you use clear enough variable names that you can put the description of the query above it.